packages/jellyfish-api-core/src/category/icxorderbook.ts

```diff
@@ -0,0 +1,381 @@
+   * @param {string} [offer.ownerAddress]           Address of DFI token and for receiving tokens in case of EXT/DFC order
+   * @param {string} [offer.receivePubkey]          Pubkey which can claim external HTLC in case of EXT/DFC order type
+   * @param {number} [order.expiry]                 Number of blocks until the offer expires, default 10 DFI blocks
+   * @param {InputUTXO[]} inputUTXOs                Specific utxos to spend
+   * @param {string} [inputUTXOs.txid]              transaction Id
+   * @param {number} [inputUTXOs.vout]              The output number
+   * @return {Promise<ICXGenericResult>}            Object indluding transaction id of the the transaction
+   */
+  async ICXMakeOffer (offer: Offer, inputUTXOs: InputUTXO[] = []): Promise<ICXGenericResult> {
+    return await this.client.call(
+      'icx_makeoffer',
+      [
+        offer, inputUTXOs
+      ],
+      'bignumber'
+    )
+  }
+
+  /**
+   * Closes offer transaction.
+   *
+   * @param {string} offerTx                        Transaction Id of maker offer
+   * @param {InputUTXO[]} inputUTXOs                Specific utxos to spend
+   * @param {string} [inputUTXOs.txid]              transaction Id
+   * @param {number} [inputUTXOs.vout]              The output number
+   * @return {Promise<ICXGenericResult>}            Object indluding transaction id of the the transaction
+   */
+  async ICXCloseOffer (offerTx: string, inputUTXOs: InputUTXO[] = []): Promise<ICXGenericResult> {
+    return await this.client.call(
+      'icx_closeoffer',
+      [
+        offerTx, inputUTXOs
+      ],
+      'bignumber'
+    )
+  }
+
+  /**
+   * Create and submits a DFC HTLC transaction
+   *
+   * @param {HTLC} htlc
+   * @param {string} [htlc.offerTx]                 Transaction Id of the offer transaction for which the HTLC is
+   * @param {BigInt} [htlc.amount]                  Amount in HTLC
+   * @param {string} [htlc.hash]                    Hash of seed used for the hash lock part
+   * @param {number} [htlc.timeout]                 Timeout (absolute in blocks) for expiration of HTLC in DFI blocks
+   * @param {InputUTXO[]} inputUTXOs                Specific utxos to spend
+   * @param {string} [inputUTXOs.txid]              transaction Id
+   * @param {number} [inputUTXOs.vout]              The output number
+   * @return {Promise<ICXGenericResult>}            Object indluding transaction id of the the transaction
+   */
+  async ICXSubmitDFCHTLC (htlc: HTLC, inputUTXOs: InputUTXO[] = []): Promise<ICXGenericResult> {
+    return await this.client.call(
+      'icx_submitdfchtlc',
+      [
+        htlc, inputUTXOs
+      ],
+      'bignumber'
+    )
+  }
+
+  /**
+   * Create and submits a external(EXT) HTLC transaction
+   *
+   * @param {ExtHTLC} htlc
+   * @param {string} [htlc.offerTx]                 Transaction Id of the offer transaction for which the HTLC is
+   * @param {BigInt} [htlc.amount]                  Amount in HTLC
```

Comment:
BigInt is not multi-platform supported it won't work on browser and React Native, use `BigNumber`.